### PR TITLE
proposal: new kurator config yaml

### PR DESCRIPTION
**Pull request by zirain**
Signed-off-by: hejianpeng <hejianpeng2@huawei.com>

**What type of PR is this?**

/kind design

**What this PR does / why we need it**:

proposal new kurator configuration file `.kurator.yaml` which locate in `$HOME/.kurator/.kurator.yaml`
this will be used to fix https://github.com/kurator-dev/kurator/issues/61 and https://github.com/kurator-dev/kurator/issues/58

1. `kurator config dump` will print into stdout
2. move `kurator tool list` to `kurator config cli list`
3. add `kurator config tool get <cli-name>`
4. plugin should get cli/charts/manifest in `Init`

cc @hzxuzhonghu @prodanlabs 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by prodanlabs** [comment]
looks good to me.

File: manifests/.kurator.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
+plugins:
+  - name: istio
+    hub: docker.io/istio
+    cli:
+      - name: istioctl
```
Comment (hzxuzhonghu) [inline manifests/.kurator.yaml:5]: why need name

File: manifests/.kurator.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  - name: kubeedge
+    hub: docker.io/kebeedge
+    cli:
+      - name: karmadactl
+  - name: volcano
+    version: 1.5.1
+    hub: docker.io/volcanosh
+    manifest:
+      - version: 1.5.1
+        releaseURLPrefix: "https://raw.githubusercontent.com/volcano-sh/volcano/"
```
Comment (hzxuzhonghu) [inline manifests/.kurator.yaml:22]: I am not sure i understand the structure

**Review by hzxuzhonghu — COMMENTED** [review]
I would suggest first designing the Schema

**Comment by prodanlabs** [comment]
The structure of the configuration file has not been decided yet?

**Comment by zirain** [comment]
> The structure of the configuration file has not been decided yet?

still need improvement, do this block your work?

**Comment by prodanlabs** [comment]
> still need improvement, do this block your work?

yes, but that's okay, has little effect.

**Pull request closed without merging**